lms/djangoapps/bookmarks/tests/test_views.py

```diff
@@ -0,0 +1,462 @@
+
+        bookmarks = response.data['results']
+
+        self.assertEqual(len(bookmarks), 2)
+        self.assertEqual(response.data['count'], 2)
+        self.assertEqual(response.data['num_pages'], 1)
+
+        # As bookmarks are sorted by -created so we will compare in that order.
+        self.assert_valid_bookmark_response(bookmarks[0], self.bookmark_2)
+        self.assert_valid_bookmark_response(bookmarks[1], self.bookmark_1)
+
+    def test_get_bookmarks_with_optional_fields(self):
+        """
+        Test that requesting bookmarks for a course returns records successfully in
+        expected order with optional fields.
+        """
+        query_parameters = 'course_id={}&fields=path,display_name'.format(self.course_id)
+
+        response = self.send_get(client=self.client, url=reverse('bookmarks'), query_parameters=query_parameters)
+        data = response.data
+        bookmarks = data['results']
+
+        self.assertEqual(len(bookmarks), 2)
+        self.assertEqual(data['count'], 2)
+        self.assertEqual(data['num_pages'], 1)
+
+        # As bookmarks are sorted by -created so we will compare in that order.
+        self.assert_valid_bookmark_response(bookmarks[0], self.bookmark_2, optional_fields=True)
+        self.assert_valid_bookmark_response(bookmarks[1], self.bookmark_1, optional_fields=True)
+
+    def test_get_bookmarks_with_pagination(self):
+        """
+        Test that requesting bookmarks for a course return results with pagination 200 code.
+        """
+        query_parameters = 'course_id={}&page_size=1'.format(self.course_id)
+        response = self.send_get(client=self.client, url=reverse('bookmarks'), query_parameters=query_parameters)
+
+        bookmarks = response.data['results']
+
+        # Pagination assertions.
+        self.assertEqual(response.data['count'], 2)
+        self.assertIn('page=2&page_size=1', response.data['next'])
+        self.assertEqual(response.data['num_pages'], 2)
+
+        self.assertEqual(len(bookmarks), 1)
+        self.assert_valid_bookmark_response(bookmarks[0], self.bookmark_2)
+
+    def test_get_bookmarks_with_invalid_data(self):
+        """
+        Test that requesting bookmarks with invalid data returns a 0 records.
+        """
+        # Invalid course id.
+        response = self.send_get(client=self.client, url=reverse('bookmarks'), query_parameters='course_id=invalid')
+        bookmarks = response.data['results']
+        self.assertEqual(len(bookmarks), 0)
+
+    def test_get_all_bookmarks_when_course_id_not_give(self):
+        """
+        Test that requesting bookmarks returns all records for that user.
+        """
+        # Without course id we would return all the bookmarks for that user.
+        response = self.send_get(client=self.client, url=reverse('bookmarks'))
+        bookmarks = response.data['results']
+        self.assertEqual(len(bookmarks), 2)
+        self.assert_valid_bookmark_response(bookmarks[0], self.bookmark_2)
```

Comment:
Need to also have bookmarks in a second course to make sure they are all returned.